src/routes/github/create-branch/github-branches-get.ts

```diff
@@ -0,0 +1,40 @@
+import { Request, Response } from "express";
+import { createUserClient } from "~/src/util/get-github-client-config";
+
+export const GithubBranchesGet = async (req: Request, res: Response): Promise<void> => {
+	const { githubToken, jiraHost, gitHubAppConfig } = res.locals;
+
+	if (!githubToken || !jiraHost || !gitHubAppConfig) {
+		res.sendStatus(401);
+		return;
+	}
+
+	const { owner, repo } = req.params;
+	if (!owner || !repo) {
+		res.status(400).json({ err: "Missing required data." });
+		return;
+	}
+
+	try {
+		const gitHubUserClient = await createUserClient(githubToken, jiraHost, req.log, gitHubAppConfig.gitHubAppId);
+		let allBranchesFetched = false;
+		let allBranches;
+		let cursor;
+		while (!allBranchesFetched) {
```

Comment:
Unsure here, probably worth a chat at tomorrows catchup!!

Just concerned if we have super large amount of branches, even 1000+ we have to wait a long time for it.

We may still land here....

Other options we can add to and discuss tomorrow:
- Get the 100 most recent branches
- Have the data streaming in behind the scenes so its non-blocking
- Get the 100 most recent branches plus a manual entry if can't find it??

Reply:
Sweet, so as I see it. we can drop the while loop and just return the 100.

We can create a seperate story to handle the UI and the branch validation if its manual.